ci/build-docker-image.sh

```diff
@@ -42,74 +41,147 @@ do
+    CI_DOCKER_BUILD_ARGS+=("--ssh=default")
+fi
+
+# Check if the specified or default Dockerfile exists
+if [[ -n "$CI_DOCKER_FILE" ]] && [[ -f "${CI_DOCKER_FILE}" ]]; then
+    CI_DOCKER_BUILD_ARGS+=("--file=${CI_DOCKER_FILE}")
+elif [[ ! -f "Dockerfile" ]]; then
+    echo -e "\e[35;1mError! No Dockerfile found.\e[0m"
+    echo -e "\e[35;1mEither provide the filepath to a Dockerfile using the arg --docker_file=/location/to/Dockerfile\e[0m"
+    echo -e "\e[35;1mor place a Dockerfile named 'Dockerfile' in the context directory.\e[0m"
+    exit 1
 fi
 
-# build the Docker image (this will use the Dockerfile in the root of the repo)
-DOCKER_BUILDKIT=1 docker build $DOCKER_SSH_ARGS --build-arg BRANCH="$CI_BRANCH" --build-arg \
-    PULL_REQUEST="$CI_PULL_REQUEST" --build-arg COMMIT="$CI_COMMIT" --build-arg \
-    CI="$CI" --build-arg BASE_IMAGE="$BASE_IMAGE" -t "$CI_DOCKER_IMAGE_NAME" .
+# Construction of Docker buildx arguments begins here
+if [[ -n "$CI_DOCKER_PLATFORMS" ]]; then
+    CI_DOCKER_BUILDX_ARGS+=("--platform=linux/${CI_DOCKER_PLATFORMS}")
 
-# push the new Docker image to the Docker registry only after acceptance of pull request
-if [ "$CI_PULL_REQUEST" == "false" ]
-then
-    # Authenticate to the Docker registry
-    echo -e "\e[35m\e[1mAuthenticating docker registry $CI_DOCKER_REGISTRY\e[0m"
-    echo "$CI_DOCKER_PASSWORD" | docker login "$CI_DOCKER_REGISTRY" -u "$CI_DOCKER_USER" --password-stdin
+    echo -e "\e[35;1mCreating a new docker context for multi arch builds\e[0m"
+    docker context create multiarch-environment
+    echo -e "\e[35;1mCreating a new buildx builder for multi arch builds\e[0m"
+    docker buildx create --name multiarchbuilder --driver docker-container --use multiarch-environment
+    docker buildx ls
+fi
+
+# Push the new Docker image to the Docker registry only after acceptance of pull/merge request
+DOCKER_PUSH=false
+if [[ "$CI_PULL_REQUEST" == "false" ]]; then
+    if [[ "$CI_DOCKER_PUSH_IMAGE" == "true" ]]; then
+        if [[ "${CI_DOCKER_LOGIN}" == "true" ]]; then
+            # Authenticate to the Docker registry
+            echo -e "\e[35;1mAuthenticating docker registry $CI_DOCKER_REGISTRY\e[0m"
+            echo "$CI_DOCKER_PASSWORD" | docker login "$CI_DOCKER_REGISTRY" -u "$CI_DOCKER_USER" --password-stdin
+        fi
+        DOCKER_PUSH=true
+    fi
+else
+    if [[ "${CI_DOCKER_PUSH_IMAGE}" == "true" ]]; then
+        echo -e "\e[33;1m--push_image=true with --pull_request=true is not allowed. Docker images are not pushed in a non-branch pipeline\e[0m"
+    fi
+fi
+CI_DOCKER_BUILDX_ARGS+=("--output=type=image,push=${DOCKER_PUSH}")
 
-    echo -e "\e[35m\e[1mdocker push $CI_DOCKER_IMAGE_NAME\e[0m"
-    docker push "$CI_DOCKER_IMAGE_NAME"
+# Build the Docker image
+echo -e "\e[35;1mBuilding docker image: ${CI_DOCKER_IMAGE_NAME} with push=${DOCKER_PUSH}\e[0m"
+echo -e "\e[36mExecuting the command: docker buildx build ${CI_DOCKER_BUILDX_ARGS[*]} ${CI_DOCKER_BUILD_ARGS[*]} .\e[0m"
```

Comment:
```suggestion
echo -e "\e[36mExecuting the command: docker buildx build ${CI_DOCKER_BUILDX_ARGS[*]} ${CI_DOCKER_BUILD_ARGS[*]}.\e[0m"
```

Reply:
The space between the `.` is needed. It is not a full-stop :smile:  Its the current directory `.`